src/argilla/client/feedback/training/frameworks/trl.py

```diff
@@ -0,0 +1,383 @@
+                ref_model=self._transformers_ref_model,
+                tokenizer=self._transformers_tokenizer,
+                dataset=dataset,
+                data_collator=data_collator,
+                **self.trainer_kwargs,
+            )
+        elif isinstance(self._task, TrainingTaskForDPO):
+            self._training_args = TrainingArguments(**self.training_args_kwargs)
+            self._trainer = self.trainer_cls(
+                model=self._transformers_model,
+                ref_model=self._transformers_ref_model,
+                args=self._training_args,
+                train_dataset=self._train_dataset,
+                eval_dataset=self._eval_dataset,
+                tokenizer=self._transformers_tokenizer,
+                **self.trainer_kwargs,
+            )
+
+        #  train
+        if isinstance(self._task, TrainingTaskForPPO):
+            import torch
+            from tqdm import tqdm
+            from trl.core import LengthSampler
+
+            output_length_sampler = LengthSampler(**self.training_args_kwargs["length_sampler_kwargs"])
+            generation_kwargs = self.training_args_kwargs["generation_kwargs"]
+            generation_kwargs["pad_token_id"] = self._transformers_tokenizer.eos_token_id
+            reward_model = self.training_args_kwargs["reward_model"]
+
+            for batch in tqdm(self._trainer.dataloader):
+                query_tensors = batch["input_ids"]
+
+                #### Get response from SFT
+                response_tensors = self._trainer.generate(
+                    query_tensors,
+                    return_prompt=False,
+                    length_sampler=output_length_sampler,
+                    **generation_kwargs,
+                )
+                batch["response"] = self._transformers_tokenizer.batch_decode(
+                    response_tensors, skip_special_tokens=True
+                )
+
+                #### Compute rewards scores
+                texts = [q + r for q, r in zip(batch["query"], batch["response"])]
+                pipe_outputs = reward_model(texts, top_k=None, truncation=True)
+                rewards = [torch.tensor(output[-1]["score"] * len(output)) for output in pipe_outputs]
+
+                #### Run PPO step
+                stats = self._trainer.step(query_tensors, response_tensors, rewards)
+                self._trainer.log_stats(stats, batch, rewards)
+        else:
+            self._trainer.train()
+            if self._trainer.eval_dataset:
+                self._metrics = self._trainer.evaluate()
+                self._logger.info(self._metrics)
+            else:
+                self._metrics = None
+
+        self.save(output_dir)
+
+    def save(self, output_dir: str) -> None:
+        """
+        Saves the model to the specified path.
+        """
+        self._transformers_model.save_pretrained(output_dir)
+        self._transformers_tokenizer.save_pretrained(output_dir)
+
```

Comment:
should we add a `push_to_hub` method to the `transformer` based models? Similarly, we can do that [for spaCy](https://huggingface.co/docs/hub/spacy#sharing-your-models) too. IMO, we would ideally align this with a nicely generated card with a link to the dataset. @dvsrepo @alvarobartt , what do you think?

Reply:
Sure, we can prepare a card once we are happy with the training interface on a separate PR if that's necessary

Reply:
Generating a card is a very nice feature, but it would have to be custom made for each of the task-framework combinations, as they all store different information (e.g. logs) and in different ways.

Reply:
I'll make an issue for the push_to_hub

Reply:
#3634 #3633